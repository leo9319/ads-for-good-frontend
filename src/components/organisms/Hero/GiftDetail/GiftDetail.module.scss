@import '../../../../styles/breakpoints';
@import '../../../../styles/colours';
@import url('../../../../styles/_spacing.scss');

.container {
  padding: var(--spacing-2xl) calc(2 * var(--spacing-5xl) + 12px);
  @media (max-width: $md-max) {
    padding: var(--spacing-2xl);
  }

  @media (max-width: $xs-max) {
    padding: 0px;
  }

  .childDetails {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    gap: var(--spacing-md);

    @media (max-width: $xs-max) {
      gap: var(--spacing-lg);
      display: flex;
      flex-direction: column;
      align-items: center;
    }

    .imageContainer {
      .imgStyle {
        width: 536px !important;
        height: 536px !important;

        @media (max-width: $md-max) {
          width: 464px !important;
          height: 464px !important;
        }

        @media (max-width: $sm-max) {
          width: 336px !important;
          height: 336px !important;
        }

        @media (max-width: $xs-max) {
          width: 100% !important;
          height: 270px !important;
        }
      }
    }

    .details {
      display: flex;
      flex-direction: column;

      @media (max-width: $xs-max) {
        padding: 0px var(--spacing-md) var(--spacing-2xl) var(--spacing-md);
      }

      .personalDetails {
        display: flex;
        flex-direction: column;
        align-items: center;
        box-sizing: border-box;
        width: 544px;

        @media (max-width: $md-max) {
          width: auto;
        }

        @media (max-width: $sm-max) {
          padding: 0px;
        }

        .names {
          display: flex;
          flex-direction: column;
          gap: var(--spacing-xs);

          @media (max-width: $xs-max) {
            gap: var(--spacing-2xs);
          }

          .dollar {
            color: $color-neutral-600;
          }
        }

        .about {
          display: flex;
          flex-flow: wrap;
          justify-content: center;
          flex-direction: column;
          padding: var(--spacing-lg);

          @media (max-width: $xs-max) {
            padding: var(--spacing-lg) 0px;
          }

          .content {
            color: $color-neutral-600;
            font-weight: var(--typography-font-weight-regular);
            font-size: var(--typography-font-size-3);
            line-height: var(--typography-line-height-3);
            text-align: center;
            text-underline-position: from-font;
            text-decoration-skip-ink: none;
            margin-bottom: var(--spacing-xs);
            width: 100%;

            @media (max-width: $xs-max) {
              font-size: var(--typography-font-size-2);
              line-height: var(--typography-line-height-2);
            }

            @media (min-width: $lg-min) {
              width: 500px;
            }
          }
        }

        .BuyGiftButton {
          background: $color-blue-500;
          width: 322px;
          height: 56px !important;
          color: $color-white;
          line-height: var(--typography-line-height-4);

          @media (max-width: $sm-max) {
            width: 336px;
            height: 44px !important;
            font-size: var(--typography-font-size-3);
            line-height: var(--typography-line-height-3);
            padding: var(--spacing-sm) calc(var(--spacing-md) + 4px);
          }

          @media (max-width: $xs-max) {
            width: 100%;
          }
        }
      }
    }
  }
}
