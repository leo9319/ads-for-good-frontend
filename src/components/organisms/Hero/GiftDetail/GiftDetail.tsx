import React, { ReactElement } from 'react';
import styles from './GiftDetail.module.scss';
import Image from '@components/atoms/Image';
import ReadMore from '@components/molecules/ReadMore';
import Button from '@radix-styles/atoms/Button';
import Text from '@radix-styles/atoms/Text';

export interface GiftDetailProps {
  /**
   * Display the Gift image, if exists
   * @property
   */
  image?: string;
  /**
   * Fallback image url for the Gift Image
   * @property
   */
  imageFallback?: string;
  /**
   * Image alt text
   * @property
   */
  imageAltText?: string;
  /**
   * Name of the Gift
   * @property
   */
  name?: string;
  /**
   * Price of the Gift
   * @property
   */
  price?: string;
  /**
   * Currency of the Gift amount
   * @property
   * @default '$'
   */
  currency?: string;
  /**
   * Details of the Gift
   * @property
   */
  details?: string;
  /**
   * Button text
   * @property
   * @default 'Buy this gift'
   */
  buttonText?: string;
  /**
   * Button onClick handler
   * @property
   */
  buttonOnClick?: () => void;
}

export const GiftDetail = ({
  image,
  imageFallback,
  imageAltText,
  name,
  price,
  currency = '$',
  details,
  buttonText = 'Buy this gift',
  buttonOnClick,
}: GiftDetailProps): ReactElement => {
  const priceDetail = `${currency}${price}`;
  return (
    <div className={styles.container}>
      <div className={styles.childDetails}>
        <div className={styles.imageContainer}>
          <Image
            className={styles.imgStyle}
            src={image}
            alt={imageAltText}
            fallbackSrc={imageFallback}
          />
        </div>
        <div className={styles.details}>
          <div className={styles.personalDetails}>
            <div className={styles.names}>
              <Text
                size={{ initial: '7', xs: '7', sm: '7', md: '8', lg: '9' }}
                weight="semibold"
                align="center"
              >
                {name}
              </Text>
              <Text
                size={{ initial: '4' }}
                weight="semibold"
                align="center"
                className={styles.dollar}
              >
                {priceDetail}
              </Text>
            </div>
            <div className={styles.about}>
              <ReadMore
                contents={[details ?? '']}
                maxLines={3}
                contentClassName={styles.content}
                align="center"
                expandButtonLabel="Show more"
                collapseButtonLabel="Show less"
              />
            </div>
            <Button
              size="xl"
              mode="secondary"
              text={buttonText}
              className={styles.BuyGiftButton}
              onClick={buttonOnClick}
            />
          </div>
        </div>
      </div>
    </div>
  );
};

export default GiftDetail;
