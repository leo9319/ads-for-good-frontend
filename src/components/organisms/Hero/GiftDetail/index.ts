export { default } from './GiftDetail';
export type { GiftDetailProps } from './GiftDetail';
export { GiftDetail } from './GiftDetail';
