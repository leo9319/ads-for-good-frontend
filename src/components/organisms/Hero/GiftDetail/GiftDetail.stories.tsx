import { Meta, StoryObj } from '@storybook/react';
import GiftDetail from './GiftDetail';
import FallbackImage from '@assets/images/FallbackImage.svg';

export const singleGiftData = {
  image:
    'https://s3-alpha-sig.figma.com/img/5dde/a548/b4bd3ee0b9db2ca002f11f8868399260?Expires=1748822400&Key-Pair-Id=APKAQ4GOSFWCW27IBOMQ&Signature=E2cuCFo0fsFFPa1fyTqZ3w9cNIEA5bGQYWYjS5flghG6-N9blzHYU-CjH2sHbATqZOR8EN6HTt1rrG0cHQQ9aO~AKdojKKcKk1852ZJTl2HMpAngvP1aaT7e3xztAky53aUsbhz7ifQg0Lm3uKdkQBX2yQ0lj1RBLhpVe~mg8y5URAb2u96PnRvNGZUtLeaumITLo9nfoajPDVk8h979REixSZZz5qXbrniYmMQN~~OxYxXzK3Zyf5cfnelq-QFE86gLdh8gIp47nFiVSxlhg7T97aeTWupuPdjRx33YfNAjVQmvD-GJGavslpJr4K2LdCH0YyWA1U2ndT37rhGdMw__',
  imageFallback: FallbackImage,
  imageAltText: 'Gift Image',
  name: 'Goats, Hens and Roosters',
  price: '150',
  details:
    'The crack of an egg. The first tiny chirp of a newborn chick. These small delights bring joy to children and families who receive hens and a rooster. Better yet, these wonderful birds provide an ongoing source of income, food rich in protein, vitamins and nutrients, and health for growing girls and boys—so their dreams can take flight.',
  currency: '$',
  buttonText: 'Buy this gift',
};

const meta = {
  title: 'Library/Organisms/Hero/GiftDetail',
  component: GiftDetail,
  parameters: {
    layout: 'fullscreen',
  },
  tags: ['autodocs'],
  argTypes: {
    image: {
      description: 'Image of the child',
      control: { type: 'text' },
    },
    imageFallback: {
      control: 'text',
    },
    imageAltText: {
      control: 'text',
    },
    name: {
      description: 'Name of the child',
      control: { type: 'text' },
    },
    price: {
      description: 'Price of the Gift',
      control: { type: 'text' },
    },
    details: {
      description: 'Details of the child',
      control: { type: 'text' },
    },
    currency: {
      description: 'Currency of the Gift amount',
      control: { type: 'text' },
    },
    buttonText: {
      description: 'Button text',
      control: { type: 'text' },
    },
    buttonOnClick: {
      description: 'Button onClick handler',
    },
  },
  excludeStories: ['singleGiftData'],
  args: {
    currency: '$',
    buttonText: 'Buy this gift',
    buttonOnClick: () => console.log('Button clicked'),
  },
} satisfies Meta<typeof GiftDetail>;

export default meta;

type Story = StoryObj<typeof meta>;

export const SingleGiftDetail: Story = {
  args: { ...singleGiftData },
};
