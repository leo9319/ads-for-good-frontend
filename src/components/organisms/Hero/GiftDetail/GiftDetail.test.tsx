import React from 'react';
import { render, screen } from '@testing-library/react';
import { GiftDetail } from './GiftDetail';
import { singleGiftData } from './GiftDetail.stories';

const defaultProps = {
  ...singleGiftData,
  buttonOnClick: jest.fn(),
};

describe('GiftDetail', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<GiftDetail {...defaultProps} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('calls buttonOnClick when button is clicked', () => {
    const button = screen.getByRole('button', {
      name: defaultProps.buttonText,
    });
    expect(button).toBeInTheDocument();
    button.click();
    expect(defaultProps.buttonOnClick).toHaveBeenCalledTimes(1);
  });
});
