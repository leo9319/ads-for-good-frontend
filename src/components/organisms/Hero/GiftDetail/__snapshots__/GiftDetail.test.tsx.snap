// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftDetail should render successfully 1`] = `
<body>
  <div>
    <div
      class="container"
    >
      <div
        class="childDetails"
      >
        <div
          class="imageContainer"
        >
          <img
            alt="Gift Image"
            class="imgWrapper imgStyle"
            src="https://s3-alpha-sig.figma.com/img/5dde/a548/b4bd3ee0b9db2ca002f11f8868399260?Expires=1748822400&Key-Pair-Id=APKAQ4GOSFWCW27IBOMQ&Signature=E2cuCFo0fsFFPa1fyTqZ3w9cNIEA5bGQYWYjS5flghG6-N9blzHYU-CjH2sHbATqZOR8EN6HTt1rrG0cHQQ9aO~AKdojKKcKk1852ZJTl2HMpAngvP1aaT7e3xztAky53aUsbhz7ifQg0Lm3uKdkQBX2yQ0lj1RBLhpVe~mg8y5URAb2u96PnRvNGZUtLeaumITLo9nfoajPDVk8h979REixSZZz5qXbrniYmMQN~~OxYxXzK3Zyf5cfnelq-QFE86gLdh8gIp47nFiVSxlhg7T97aeTWupuPdjRx33YfNAjVQmvD-GJGavslpJr4K2LdCH0YyWA1U2ndT37rhGdMw__"
          />
        </div>
        <div
          class="details"
        >
          <div
            class="personalDetails"
          >
            <div
              class="names"
            >
              <span
                class="rt-Text rt-r-ta-center mocked-class-name typography-font-family-default"
                id=""
              >
                Goats, Hens and Roosters
              </span>
              <span
                class="rt-Text rt-r-ta-center mocked-class-name typography-font-family-default dollar"
                id=""
              >
                $150
              </span>
            </div>
            <div
              class="about"
            >
              <div>
                <div
                  class="content"
                  style="max-height: none; overflow: hidden; transition: max-height 0.3s ease; display: block;"
                >
                  The crack of an egg. The first tiny chirp of a newborn chick. These small delights bring joy to children and families who receive hens and a rooster. Better yet, these wonderful birds provide an ongoing source of income, food rich in protein, vitamins and nutrients, and health for growing girls and boys—so their dreams can take flight.
                </div>
              </div>
            </div>
            <button
              aria-label="Buy this gift"
              class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-Button wvButton xl secondary none BuyGiftButton"
              data-accent-color=""
            >
              <div
                class="buttonContent textOnly"
              >
                <span>
                  Buy this gift
                </span>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
